packages/vsce/src/commands/showLogsCommand.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { commands, TreeView, window } from "vscode";
+import { CICSRegionTree } from "../trees/CICSRegionTree";
+import { findSelectedNodes } from "../utils/commandUtils";
+import { ProfileManagement } from "../utils/profileManagement";
+import { Gui } from "@zowe/zowe-explorer-api";
+import { getResource } from "@zowe/cics-for-zowe-sdk";
+import * as vscode from "vscode";
+import { IProfileLoaded } from "@zowe/imperative";
+
+export async function findRelatedZosProfiles(cicsProfile: IProfileLoaded, zosProfiles: IProfileLoaded[]): Promise<IProfileLoaded> {
+  const baseForCicsProfile = await ProfileManagement.getProfilesCache().fetchBaseProfile(cicsProfile.name);
+
+  // find all zosmf and rse profiles (put zosmf first)
+  zosProfiles = zosProfiles.sort((prof) => (prof.profile.type === "zosmf" ? -1 : 1));
+
+  // filter out profiles that are not in the same base as the cics profile
+  const matchingProfiles: IProfileLoaded[] = [];
+  if (baseForCicsProfile) {
+    for (const profile of zosProfiles) {
+      if (baseForCicsProfile && baseForCicsProfile?.name === (await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name))?.name) {
+        matchingProfiles.push(profile);
+      }
+    }
+  }
```

Comment:
@JillieBeanSim / @zFernand0 - this is the part of the PR I'm particularly interested in your thoughts on.

I'm taking a CICS profile and a list of proposed zosProfiles.

I want to identify a zosProfile that matches the CICS profile. I'm currently doing this by requesting the base profile for each proposed zosProfile to see if it matches the base profile of the cicsProfile.

Is this the best way, currently?

Reply:
For Profiles that match a particular CICS profile you may want to use [ProfilesCache.fetchAllProfiles()](https://github.com/zowe/zowe-explorer-vscode/blob/a05967b41a770e0ec2542a7b686c3a1c0e506b03/packages/zowe-explorer-api/src/profiles/ProfilesCache.ts#L278) then filter down to the profiles with matching host.  Also, depending on the scenario you may want to filter even further on profile types you may want to support removing types like TSO, SSH or what doesn't fit.  All these profiles will already have any merging done if needed, relying on base profile can prove to be troublesome since users configure their Zowe Config in different ways and sometimes without base profiles.

Reply:
if you know the profile type there is also [ProfilesCache.fetchAllProfilesByType(type)](https://github.com/zowe/zowe-explorer-vscode/blob/a05967b41a770e0ec2542a7b686c3a1c0e506b03/packages/zowe-explorer-api/src/profiles/ProfilesCache.ts#L259) returning complete/merged profiles